modules/gapi/src/backends/streaming/gstreamingkernel.hpp

```diff
@@ -0,0 +1,37 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2020 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_GSTREAMINGKERNEL_HPP
+#define OPENCV_GAPI_GSTREAMINGKERNEL_HPP
+
+#include "compiler/gislandmodel.hpp"
+
+namespace cv {
+namespace gapi {
+namespace streaming {
+
+class IActor {
+public:
+    using Ptr = std::shared_ptr<IActor>;
+
+    virtual void run(cv::gimpl::GIslandExecutable::IInput  &in,
+                     cv::gimpl::GIslandExecutable::IOutput &out) = 0;
+
+    virtual ~IActor() = default;
+};
+
+using CreateActorFunction = std::function<IActor::Ptr()>;
+struct GStreamingKernel
+{
+    CreateActorFunction createActorFunction;
+};
+
+} // namespace streaming
+} // namespace gapi
+} // namespace cv
+
+#endif // OPENCV_GAPI_GSTREAMINGKERNEL_HPP
```

Comment:
Not sure if you need this separate header here. Any plans to include it from more than 1 file?

Reply:
I would keep the header for now. But yeah, it's only included in `gstreamingbackend.hpp`

Reply:
I see it was also reused in the /internal/ backends.. I commented it there.